infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProject.java

```diff
@@ -133,15 +134,12 @@ void deleteIfManaged() throws InfrastructureException {
 
     OpenShiftClient osClient = clientFactory.createOC(workspaceId);
 
-    if (!isManagedInternal(osClient)) {
-      throw new InfrastructureException(
-          format(
-              "Can't delete project '%s' that contains"
-                  + " runtime of workspace '%s' because it doesn't have the '"
-                  + MANAGED_NAMESPACE_LABEL
-                  + "' label equal to 'true'.",
-              projectName,
-              workspaceId));
+    if (!isProjectManaged(osClient)) {
```

Comment:
I would negate the logic to more straight-forward:
```
if managed {
  delete
} else {
  LOG
}
```

Reply:
I prefer less nesting. The `if (!is*Managed()` can be thought of as a precondition to the actual delete.